artiq/firmware/runtime/main.rs

```diff
@@ -151,12 +153,22 @@ fn startup() {
     } else {
         false
     };
-    let interface = smoltcp::iface::InterfaceBuilder::new(net_device, vec![])
+    let mut interface = smoltcp::iface::InterfaceBuilder::new(net_device, vec![])
         .hardware_addr(HardwareAddress::Ethernet(net_addresses.hardware_addr))
         .init_ip_addrs(&net_addresses)
         .neighbor_cache(neighbor_cache)
+        .routes(Routes::new(BTreeMap::new()))
         .finalize();
 
+    if !use_dhcp {
+        if let Some(default_route) = net_addresses.ipv4_default_route {
+            interface.routes_mut().add_default_ipv4_route(default_route).unwrap();
+        }
+        if let Some(default_routev6) = net_addresses.ipv6_default_route {
+            interface.routes_mut().add_default_ipv6_route(default_routev6).unwrap();
+        }
```

Comment:
I think we should probably be setting this even if ipv4 is set to use DHCP. There's no DHCPv6 support and I think that would be separate. (There doesn't seem to be DHCPv6 support in smoltcp yet).